### ISSUE TITLE
What does Typescript do that doesn't make Babel? Requesting Typed Functions and variables 

### ISSUE THREAD

**Issue by chiqui3d** [babel/babel#8637]
I think you should implement the typing of functions and variables yourselves and it doesn't depend on .ts files. 

My request is that. I hope you can implement it in addition to the decorations of functions and classes.

That would be nice to program and do things right. 
For me now is mounted a chaos rather than a solution in the javascript environment.
It can't be that you need 500 libraries to do things right. Or that to make a simple hello world! in javascript I need to know NPM, Webpack, HTML, CSS(sass or less) Babel, Javascript, Typescript, (Angular, React or Vue).I've seen several websites with several frameworks inserted, because if you don't understand the subject, in the end you create a CHAOS of programming.

**Comment by nicolo-ribaudo** [comment]
Are you proposing that Babel should type-check your code? That is not a goal of Babel (which is just a compiler), and it would require too much work (we are a quite small team with limited resources).

**Comment by chiqui3d** [comment]
Yes exactly that, currently thanks to you I can work correctly with Javascript applying good practices, having that option as optional would be ideal.

**Comment by milesj** [comment]
I don't speak for Babel officially, but this will never happen.
